Source/WebInspectorUI/UserInterface/Controllers/NetworkManager.js

```diff
@@ -992,19 +992,20 @@ WI.NetworkManager = class NetworkManager extends WI.Object
                 return;
 
             case WI.LocalResourceOverride.InterceptType.Request: {
+                let method = localResource.requestMethod ?? (isPassthrough ? request.method : "");
                 target.NetworkAgent.interceptWithRequest.invoke({
                     requestId,
                     url: localResourceOverride.generateRequestRedirectURL(request.url) ?? undefined,
-                    method: localResource.requestMethod ?? (isPassthrough ? request.method : ""),
+                    method,
                     headers: {...originalHeaders, ...localResource.requestHeaders},
                     postData: (function() {
-                        if (!WI.HTTPUtilities.RequestMethodsWithBody.has(localResource.requestMethod))
-                            return undefined;
-                        if (localResource.requestData ?? false)
-                            return btoa(localResource.requestData);
-                        if (isPassthrough)
-                            return request.data;
-                        return "";
+                        if (method && WI.HTTPUtilities.RequestMethodsWithBody.has(method)) {
```

Comment:
Wouldn't returning early here be possible and also be the preferred style?

Reply:
i didn't like the idea of having 2 different `return undefined;`